pkg/registry-disk/registry-disk.go

```diff
@@ -0,0 +1,199 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package registrydisk
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/jeevatkm/go-model"
+
+	kubev1 "k8s.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+const registryDiskV1Alpha = "ContainerRegistryDisk:v1alpha"
+const defaultIqn = "iqn.2017-01.io.kubevirt:wrapper/1"
+const defaultPort = 3261
+const defaultPortStr = "3261"
+const defaultHost = "127.0.0.1"
+
+func DisksAreReady(pod *kubev1.Pod) bool {
+	// Wait for readiness probes on image wrapper containers
+	for _, containerStatus := range pod.Status.ContainerStatuses {
+		if strings.Contains(containerStatus.Name, "disk") == false {
+			// only check readiness of disk containers
+			continue
+		}
+		if containerStatus.Ready == false {
+			return false
+		}
+	}
+	return true
+}
+
+// The virt-handler converts registry disks to their corresponding iscsi network
+// disks when the VM spec is being defined as a domain with libvirt.
+// The ports and host of the iscsi disks are already provided here by the controller.
+func MapRegistryDisks(vm *v1.VM) (*v1.VM, error) {
+	vmCopy := &v1.VM{}
+	model.Copy(vmCopy, vm)
```

Comment:
Hm, at some point, it is better, to just copy the vm object once, when we read it the first time out of the cache in the loop.